http/interceptor.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+// rwInterceptor intercepts the ResponseWriter, so it can track response size
+// and returned status code.
+type rwInterceptor struct {
+	w           http.ResponseWriter
+	tx          types.Transaction
+	headersSent bool
+}
+
+func (i *rwInterceptor) WriteHeader(statusCode int) {
+	if i.headersSent {
+		return
+	}
+
+	for k, vv := range i.w.Header() {
+		for _, v := range vv {
+			i.tx.AddResponseHeader(k, v)
+		}
+	}
+
+	i.headersSent = true
+	if it := i.tx.ProcessResponseHeaders(statusCode, "http/1.1"); it != nil {
+		processInterruption(i.w, it)
+		return
+	}
+
+	i.w.WriteHeader(statusCode)
+}
+
+func (i *rwInterceptor) Write(b []byte) (int, error) {
+	return i.tx.ResponseBodyWriter().Write(b)
+}
+
+func (i *rwInterceptor) Header() http.Header {
+	return i.w.Header()
+}
+
+// Heavily inspired in https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/server.go#L218
+func (r *rwInterceptor) wrap() http.ResponseWriter { // nolint:gocyclo
+	var (
+		hj, i0 = r.w.(http.Hijacker)
```

Comment:
`a, isA :=` pattern would probably help a lot with readability of this

Reply:
Good idea. Done.